### ISSUE TITLE
Original (already saved) file and configuration settings lost after loss of power

### ISSUE THREAD

**Issue by jld123** [notepad-plus-plus/notepad-plus-plus#2381]
### Description of the Issue

File was open in editor. Put notebook in standby and left. Battery ran out of power and notebook went off. After my return I started up the notebook again. Notepad++ lost its configuration settings. But even more importanly, the file that was open in the editor only contained NUL NUL NUL .... Not only the differences to the last save point were lost but the entire file! This has been an issue for years (google notepad++ nul nul nul). 
### Steps to Reproduce the Issue

Tried to reproduce the issue by forcefully restarting shutting off the notebook (removed battery and pulled the plug), but that did not work. It does not seem to happen every time or only under certain circumstances

Sorry that I can't provide more information on how reproduce this bug. However, it seems to have been an issue for years. And the bug is of such a high severity (the original saved file should NEVER be destroyed) that I still decided to report this.

**Comment by marcusfr123** [comment]
Even if this bug is not fixed it would be helpful to have infos under which circumstances this happens, what exactly happens (e.g. are all files open in the editor affected or only the one being edited or only changed files or...) and how to mitigate the problem (e.g. are notepad++ simple/verbose/session backup files also affected or can the original file be restored through those?)

**Comment by SD-DAken** [comment]
The problem still exists in Notepad++ 7.5.1

The whole computer was hung (probably not Notepad++ 's fault) and had to be forcefully turned off, after restart the whole file that was currently open in Notepad++ was overwritten with NUL (0x00 bytes).

The file size seemed to be correct but every byte was replaced by 0x00 (NUL).

Even though no exact steps to reproduce exist the issue is severe enough that the code that saves files should be checked if there are hints what could result in such catastrophic failure.

I could understand if the unsaved changes are lost or the file ends up in an inconstent state if the programm / pc crashes, but the whole file being replaced by null bytes is unexpected and the worst that could possibly happen.

**Comment by pryrt** [comment]
It's come up yet again in the [forums](https://notepad-plus-plus.org/community/topic/13302/fix-corrupted-txt-file-null/17).  

At one point last month, I tried using the windows API from an external process to try to kill the Notepad++ instance, and see if I could reliably replicate the issue, but I couldn't get it to trigger; apparently, even killing the process externally is too "clean" of an exit.

Today, I began to think that maybe there was a way to force a Windows Blue Screen Of Death while running NPP, and see if it could replicate the problem -- my thought was in a Virtual Machine, to protect any actual hardware.  

I found [this MS TechNet post](https://social.technet.microsoft.com/wiki/contents/articles/6529.how-to-manually-initiate-a-windows-stop-error-bsod-and-create-a-dump-file.aspx) which explains how to initiate a BSOD.  It has a disclaimer, "you cannot manually initiate a Stop error on a virtual machine that has virtual machine extensions installed", but that implies that a VM _without_ extensions installed _should_ be able to be forced to BSOD.

I don't have a VM (or a physical machine that I could risk), nor do I have the windows-app debugging skill necessary to figure out what's going wrong, even if I could replicate the problem reliably... but I thought I'd share the idea, in case someone else has the right tools and skillset.
